File: app/src/main/java/com/kode/utair/presentation/ui/activities/WeatherForecastActivity.java
package com.kode.utair.presentation.ui.activities;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import androidx.viewpager.widget.ViewPager;
import androidx.appcompat.widget.Toolbar;
import android.view.View;
import android.widget.TextView;

import com.arellomobile.mvp.presenter.InjectPresenter;
import com.arellomobile.mvp.presenter.ProvidePresenter;
import com.kode.utair.R;
import com.kode.utair.UTairApplication;
import com.kode.utair.presentation.mvp.presenters.WeatherForecastPresenter;
import com.kode.utair.presentation.mvp.views.WeatherForecastView;
import com.kode.utair.presentation.ui.adapters.CitiesForecastPagerAdapter;

public class WeatherForecastActivity extends BaseMvpActivity implements WeatherForecastView {

    public static final int DEPART_CITY_TAB_POSITION = 0;
    public static final int ARRIVE_CITY_TAB_POSITION = 1;

    public static Intent buildIntent(Activity activity) {
        return new Intent(activity, WeatherForecastActivity.class);
    }

    @InjectPresenter
    WeatherForecastPresenter weatherPresenter;

    @ProvidePresenter
    WeatherForecastPresenter providePresenter() {
        return new WeatherForecastPresenter(UTairApplication.component());
    }

    private TextView departCity;
    private TextView arriveCity;
    private View forwardTab;
    private View returnTab;
    private ViewPager citiesForecastPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_weather_forecast);

        initToolbar();
        initViews();
    }

    private void initToolbar() {
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        toolbar.setNavigationOnClickListener(v -> onBackPressed());
    }

    private void initViews() {
        departCity = (TextView) findViewById(R.id.depart_city);
        arriveCity = (TextView) findViewById(R.id.arrive_city);
        forwardTab = findViewById(R.id.forward_tab);
        returnTab = findViewById(R.id.return_tab);
        forwardTab.setOnClickListener(v -> weatherPresenter.onTabSelected(DEPART_CITY_TAB_POSITION));
        returnTab.setOnClickListener(v -> weatherPresenter.onTabSelected(ARRIVE_CITY_TAB_POSITION));

        citiesForecastPager = (ViewPager) findViewById(R.id.cities_forecasts_pager);
        citiesForecastPager.addOnPageChangeListener(new ViewPager.SimpleOnPageChangeListener() {
            @Override
            public void onPageSelected(int position) {
                weatherPresenter.onTabSelected(position);
            }
        });
    }

    @Override
    public void showCitiesNames(String departCityName, String arriveCityName) {
        departCity.setText(departCityName);
        arriveCity.setText(arriveCityName);
    }

    @Override
    public void setTabSelected(int currentTabPosition) {
        forwardTab.setSelected(currentTabPosition == DEPART_CITY_TAB_POSITION);
        returnTab.setSelected(currentTabPosition == ARRIVE_CITY_TAB_POSITION);
    }

    @Override
    public void openForecastPage(int position) {
        citiesForecastPager.setCurrentItem(position);
    }

    @Override
    public void showForecastForCities(String departCity, String arriveCity) {
        citiesForecastPager.setAdapter(new CitiesForecastPagerAdapter(getSupportFragmentManager(), departCity, arriveCity));
    }


}

File: app/src/main/java/com/kode/utair/presentation/ui/activities/WeatherForecastActivity.kt
package com.kode.utair.presentation.ui.activities

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import androidx.viewpager.widget.ViewPager.SimpleOnPageChangeListener
import com.arellomobile.mvp.presenter.InjectPresenter
import com.arellomobile.mvp.presenter.ProvidePresenter
import com.kode.utair.R
import com.kode.utair.UTairApplication.Companion.component
import com.kode.utair.presentation.mvp.presenters.WeatherForecastPresenter
import com.kode.utair.presentation.mvp.views.WeatherForecastView
import com.kode.utair.presentation.ui.adapters.CitiesForecastPagerAdapter
import kotlinx.android.synthetic.main.activity_weather_forecast.*
import kotlinx.android.synthetic.main.cities_forecasts_tabs.*
import kotlinx.android.synthetic.main.toolbar_white.*

class WeatherForecastActivity : BaseMvpActivity(), WeatherForecastView {

    @InjectPresenter
    lateinit var weatherPresenter: WeatherForecastPresenter

    @ProvidePresenter
    fun providePresenter(): WeatherForecastPresenter {
        return component().getWeatherForecastPresenter()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_weather_forecast)

        toolbar.setNavigationOnClickListener { onBackPressed() }
        forwardTab.setOnClickListener {
            weatherPresenter.onTabSelected(DEPART_CITY_TAB_POSITION)
        }
        returnTab.setOnClickListener {
            weatherPresenter.onTabSelected(ARRIVE_CITY_TAB_POSITION)
        }
        citiesForecastPager.addOnPageChangeListener(object : SimpleOnPageChangeListener() {
            override fun onPageSelected(position: Int) {
                weatherPresenter.onTabSelected(position)
            }
        })
    }

    override fun showCitiesNames(departCityName: String, arriveCityName: String) {
        this@WeatherForecastActivity.departCityLabel.text = departCityName
        this@WeatherForecastActivity.arriveCityLabel.text = arriveCityName
    }

    override fun setTabSelected(currentTabPosition: Int) {
        forwardTab.isSelected = currentTabPosition == DEPART_CITY_TAB_POSITION
        returnTab.isSelected = currentTabPosition == ARRIVE_CITY_TAB_POSITION
    }

    override fun openForecastPage(position: Int) {
        citiesForecastPager.currentItem = position
    }

    override fun showForecastForCities(departCity: String, arriveCity: String) {
        citiesForecastPager.adapter = CitiesForecastPagerAdapter(
                fragmentManager = supportFragmentManager,
                departCityName = departCity,
                arriveCityName = arriveCity
        )
    }

    companion object {
        const val DEPART_CITY_TAB_POSITION = 0
        const val ARRIVE_CITY_TAB_POSITION = 1
        fun buildIntent(activity: Activity?): Intent {
            return Intent(activity, WeatherForecastActivity::class.java)
        }
    }
}